File: screenshots/capture.js
// Screenshot capture script using Chrome DevTools Protocol
// Usage: /Applications/Google\ Chrome.app/.../Google\ Chrome --headless --disable-gpu --run-all-compositor-stages-before-draw ...
// We'll use a simpler approach: node with Chrome's built-in fetch + screencapture

const { execSync } = require('child_process');
const http = require('http');
const https = require('https');
const fs = require('fs');
const path = require('path');

const CHROME = '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome';
const BASE = 'http://localhost:4000';
const OUT = __dirname;

// Chrome headless screenshot function
function screenshot(url, filename, width = 1280, height = 900) {
  const outPath = path.join(OUT, filename);
  try {
    execSync(
      `"${CHROME}" --headless=new --disable-gpu --screenshot="${outPath}" --window-size=${width},${height} --hide-scrollbars --default-background-color=0 "${url}"`,
      { timeout: 30000, stdio: 'pipe' }
    );
    console.log(`  OK: ${filename}`);
  } catch (e) {
    console.error(`  FAIL: ${filename} — ${e.message.slice(0, 100)}`);
  }
}

// For authenticated pages we need cookies. Let's get a session cookie first.
function getSessionCookie(email, password) {
  return new Promise((resolve, reject) => {
    // First, get the CSRF token from the sign-in page
    const getReq = http.get(`${BASE}/sign-in`, (res) => {
      let body = '';
      // Grab the set-cookie headers
      const cookies = (res.headers['set-cookie'] || []).map(c => c.split(';')[0]).join('; ');

      res.on('data', d => body += d);
      res.on('end', () => {
        // Find CSRF token
        const csrfMatch = body.match(/name="_csrf_token"[^>]*value="([^"]+)"/);
        if (!csrfMatch) {
          // Try the other format
          const altMatch = body.match(/csrf_token.*?value="([^"]+)"/s);
          if (!altMatch) {
            console.log('Could not find CSRF token, trying form post anyway...');
          }
        }
        const csrf = csrfMatch ? csrfMatch[1] : '';

        // POST to sign-in
        const postData = `customer[email]=${encodeURIComponent(email)}&customer[password]=${encodeURIComponent(password)}&_csrf_token=${encodeURIComponent(csrf)}`;

        const postReq = http.request(`${BASE}/auth/customer/password/sign_in`, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/x-www-form-urlencoded',
            'Cookie': cookies
          }
        }, (postRes) => {
          const sessionCookies = (postRes.headers['set-cookie'] || []).map(c => c.split(';')[0]).join('; ');
          resolve(sessionCookies || cookies);
        });

        postReq.on('error', reject);
        postReq.write(postData);
        postReq.end();
      });
    });
    getReq.on('error', reject);
  });
}

async function main() {
  console.log('Taking screenshots...\n');

  // Public pages (no auth needed)
  console.log('--- Public Pages ---');
  screenshot(`${BASE}/`, '01_landing.png', 1280, 1600);
  screenshot(`${BASE}/book`, '02_booking_step1.png');
  screenshot(`${BASE}/sign-in`, '03_signin.png');
  screenshot(`${BASE}/book/success`, '04_payment_success.png');
  screenshot(`${BASE}/book/cancel`, '05_payment_cancel.png');

  // For authenticated pages, Chrome headless can't easily pass session cookies
  // Let's use a different approach: create a small HTML page that logs in via JS and navigates

  // Actually, Chrome headless supports --user-data-dir with pre-set cookies
  // Simpler: we'll use the dev routes bypass if available, or screenshot what we can

  // Let's check if dev routes are enabled (they should be in dev mode)
  console.log('\n--- Customer Pages (need auth) ---');
  screenshot(`${BASE}/appointments`, '06_appointments.png');

  console.log('\n--- Technician Pages (need auth) ---');
  screenshot(`${BASE}/tech`, '07_tech_dashboard.png');

  console.log('\n--- Admin Pages (need auth) ---');
  screenshot(`${BASE}/admin/dispatch`, '08_dispatch.png');
  screenshot(`${BASE}/admin/metrics`, '09_metrics.png');
  screenshot(`${BASE}/admin/events`, '10_events.png');
  screenshot(`${BASE}/admin/formation`, '11_formation.png');
  screenshot(`${BASE}/admin/org-chart`, '12_org_chart.png');
  screenshot(`${BASE}/admin/procedures`, '13_procedures.png');

  console.log('\nDone! Check screenshots/ directory.');
}

main().catch(console.error);

File: screenshots/capture.py
#!/usr/bin/env python3
"""Capture screenshots of all app screens using Chrome headless."""

import subprocess
import os
import time

CHROME = "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"
BASE = "http://localhost:4000"
OUT = os.path.dirname(os.path.abspath(__file__))

def screenshot(url, filename, width=1280, height=900):
    outpath = os.path.join(OUT, filename)
    cmd = [
        CHROME,
        "--headless=new",
        "--disable-gpu",
        "--no-sandbox",
        "--disable-web-security",
        "--hide-scrollbars",
        f"--screenshot={outpath}",
        f"--window-size={width},{height}",
        url
    ]
    try:
        subprocess.run(cmd, timeout=30, capture_output=True)
        if os.path.exists(outpath):
            size = os.path.getsize(outpath)
            print(f"  OK: {filename} ({size:,} bytes)")
            return True
        else:
            print(f"  FAIL: {filename} — no file created")
            return False
    except Exception as e:
        print(f"  FAIL: {filename} — {e}")
        return False

def main():
    print("Capturing screenshots...\n")

    # Public pages
    print("--- Public Pages ---")
    screenshot(f"{BASE}/", "01_landing.png", 1280, 2000)
    screenshot(f"{BASE}/book", "02_booking_step1.png", 1280, 1000)
    screenshot(f"{BASE}/sign-in", "03_signin.png", 1280, 800)
    screenshot(f"{BASE}/book/success", "04_payment_success.png", 1280, 600)
    screenshot(f"{BASE}/book/cancel", "05_payment_cancel.png", 1280, 600)

    # Customer pages (dev_routes bypasses auth)
    print("\n--- Customer Pages ---")
    screenshot(f"{BASE}/appointments", "06_appointments.png", 1280, 1000)
    # We can't easily get a specific appointment ID, but we can try
    screenshot(f"{BASE}/appointments", "06b_appointments_detail.png", 390, 844)  # mobile view

    # Technician pages
    print("\n--- Technician Pages ---")
    screenshot(f"{BASE}/tech", "07_tech_dashboard.png", 390, 844)  # mobile-first
    screenshot(f"{BASE}/tech", "07b_tech_dashboard_desktop.png", 1280, 1000)

    # Admin pages
    print("\n--- Admin Pages ---")
    screenshot(f"{BASE}/admin/dispatch", "08_dispatch.png", 1280, 1200)
    screenshot(f"{BASE}/admin/metrics", "09_metrics.png", 1280, 1800)
    screenshot(f"{BASE}/admin/events", "10_events.png", 1280, 1000)
    screenshot(f"{BASE}/admin/formation", "11_formation.png", 1280, 1600)
    screenshot(f"{BASE}/admin/org-chart", "12_org_chart.png", 1280, 1000)
    screenshot(f"{BASE}/admin/procedures", "13_procedures.png", 1280, 1400)

    print(f"\nDone! Screenshots saved to {OUT}/")

if __name__ == "__main__":
    main()
